plugin.js

```diff
@@ -53,7 +53,6 @@ function plugin (fastify, options, pluginRegistrationDone) {
     req.log.trace('sessionId: %s', sessionId)
     if (sessionId === false) {
       req.log.warn('session id signature mismatch, starting new session')
-      req.session = getSession()
       return hookFinished()
     }
```

Comment:
This shouldn't be elevated out of the `if` block. This is resetting the session to an empty session because signature did not match.